fe/fe-core/src/main/java/com/starrocks/lake/snapshot/ClusterSnapshotJobScheduler.java

```diff
@@ -0,0 +1,119 @@
+            CheckpointController starMgrController) {
+        super("cluster_snapshot_checkpoint_scheduler", 10L);
+        this.feController = feController;
+        this.starMgrController = starMgrController;
+        this.restoredSnapshotInfo = RestoreClusterSnapshotMgr.getRestoredSnapshotInfo();
+        this.lastAutomatedJobStartTimeMs = 0;
+    }
+
+    @Override
+    public CheckpointController getFeController() {
+        return feController;
+    }
+
+    @Override
+    public CheckpointController getStarMgrController() {
+        return starMgrController;
+    }
+
+    @Override
+    public Pair<Long, Long> captureConsistentCheckpointIdBetweenFEAndStarMgr() {
+        if (feController == null || starMgrController == null) {
+            return null;
+        }
+
+        int retryTime = CAPTURE_ID_RETRY_TIME;
+        while (retryTime > 0) {
+            long feCheckpointIdT1 = feController.getJournal().getMaxJournalId();
+            long starMgrCheckpointIdT2 = starMgrController.getJournal().getMaxJournalId();
+            long feCheckpointIdT3 = feController.getJournal().getMaxJournalId();
+            long starMgrCheckpointIdT4 = starMgrController.getJournal().getMaxJournalId();
+
+            if (feCheckpointIdT1 == feCheckpointIdT3 && starMgrCheckpointIdT2 == starMgrCheckpointIdT4) {
+                return Pair.create(feCheckpointIdT3, starMgrCheckpointIdT2);
+            }
+
+            try {
+                Thread.sleep(100);
+            } catch (Exception ignore) {
+            }
+            --retryTime;
+        }
+        return null;
+    }
+
+    @Override
+    protected void runAfterCatalogReady() {
+        // skip first run when the scheduler start
+        if (lastAutomatedJobStartTimeMs == 0) {
+            GlobalStateMgr.getCurrentState().getClusterSnapshotMgr()
+                    .resetSnapshotJobsStateAfterRestarted(restoredSnapshotInfo);
+            lastAutomatedJobStartTimeMs = System.currentTimeMillis(); // init last start time
+            return;
+        }
+
+        /*
+         * Control the interval of automated cluster snapshot manually instead of by
+         * Daemon framework
+         * for the future purpose.
+         */
+        if (!GlobalStateMgr.getCurrentState().getClusterSnapshotMgr().canScheduleNextJob(lastAutomatedJobStartTimeMs)
+                || (runningJob != null && runningJob.isUnFinishedState())) {
+            return;
+        }
+
+        CheckpointController.exclusiveLock();
+        try {
+            if (runningJob == null || !runningJob.isUnFinishedState()) {
+                runningJob = GlobalStateMgr.getCurrentState().getClusterSnapshotMgr().getNextCluterSnapshotJob();
+            }
+
+            // set last start time when job has been created and begin to submit
+            lastAutomatedJobStartTimeMs = runningJob.getCreatedTimeMs();
+            runningJob.run(this);
+        } finally {
+            if (runningJob != null && !runningJob.isUnFinishedState()) {
```

Comment:
Why not `runningJob = null;` if `!runningJob.isUnFinishedState()`? In this case, it seems that the job has encounter some error?